command/volume_register_test.go

```diff
@@ -42,60 +42,90 @@ rando = "bar"
 	}
 }
 
-func TestCSIVolumeParse(t *testing.T) {
+func TestCSIVolumeDecode(t *testing.T) {
 	t.Parallel()
 
 	cases := []struct {
-		hcl string
-		q   *api.CSIVolume
-		err string
+		name     string
+		hcl      string
+		expected *api.CSIVolume
+		err      string
 	}{{
 		hcl: `
-id = "foo"
-type = "csi"
-namespace = "n"
-access_mode = "single-node-writer"
-attachment_mode = "file-system"
-plugin_id = "p"
+id              = "testvolume"
+name            = "test"
+type            = "csi"
+plugin_id       = "myplugin"
+
+capacity_min = "10 MiB"
+capacity_max = "1G"
+snapshot_id  = "snap-12345"
+
+mount_options {
+  fs_type     = "ext4"
+  mount_flags = ["ro"]
+}
+
 secrets {
-  mysecret = "secretvalue"
+  password = "xyzzy"
+}
+
+parameters {
+  skuname = "Premium_LRS"
+}
+
+capability {
+  access_mode     = "single-node-writer"
+  attachment_mode = "file-system"
+}
+
+capability {
+  access_mode     = "single-node-reader-only"
+  attachment_mode = "block-device"
 }
 `,
-		q: &api.CSIVolume{
-			ID:             "foo",
-			Namespace:      "n",
-			AccessMode:     "single-node-writer",
-			AttachmentMode: "file-system",
-			PluginID:       "p",
-			Secrets:        api.CSISecrets{"mysecret": "secretvalue"},
-		},
-		err: "",
-	}, {
-		hcl: `
-{"id": "foo", "namespace": "n", "type": "csi", "access_mode": "single-node-writer", "attachment_mode": "file-system",
-"plugin_id": "p"}
-`,
-		q: &api.CSIVolume{
-			ID:             "foo",
-			Namespace:      "n",
-			AccessMode:     "single-node-writer",
-			AttachmentMode: "file-system",
-			PluginID:       "p",
+		name: "typical volume",
```

Comment:
Starting the case with name makes it more obvious for the reader the the test case hcl content is going to be about.